src/modules/post/repository.rs

```diff
@@ -33,32 +37,51 @@ impl PostRepository {
         Self { database }
     }
 
-    pub async fn insert(&self, user: User, content: &str) -> Result<Post> {
-        let conn = self.database.conn_pool.get()?;
-        let dto = InsertPostTableRow {
-            content: content.to_string(),
-            user_id: user.id.clone(),
-        };
-        let row = diesel::insert_into(posts::table)
-            .values(dto)
-            .get_result::<PostsTableRow>(&conn)?;
-        let post = Post {
-            id: row.id,
-            content: row.content,
-            author: user,
-            created_at: row.created_at,
-            updated_at: row.updated_at,
-        };
-
-        Ok(post)
+    pub async fn find_by_author(&self, author_id: &Uuid) -> Result<Vec<PostsTableRow>> {
+        let result: Vec<PostsTableRow> = sqlx::query_as("SELECT * FROM posts WHERE user_id = $1")
+            .bind(author_id)
+            .fetch_all(&self.database.conn_pool)
+            .await?;
+        let posts = result
+            .into_iter()
+            .map(|row| PostsTableRow {
+                id: row.id,
+                user_id: row.user_id,
+                content: row.content,
+                scope: row.scope,
+                created_at: row.created_at,
+                updated_at: row.updated_at,
+            })
+            .collect::<Vec<PostsTableRow>>();
+        Ok(posts)
     }
 
-    pub async fn find_by_author(&self, author_id: &Uuid) -> Result<Vec<PostsTableRow>> {
-        let conn = self.database.conn_pool.get()?;
-        let posts = posts::table
-            .filter(posts::user_id.eq(author_id))
-            .load::<PostsTableRow>(&conn)?;
+    pub async fn insert(&self, user: User, dto: InsertPostTableRow) -> Result<Post> {
+        let result: PostsTableRow = sqlx::query_as(
+            r#"
+            INSERT INTO posts (
+                content,
+                scope,
+                user_id
+            ) VALUES (
+                $1,
+                $2::scope,
+                $3
+            ) RETURNING *"#,
+        )
+        .bind(dto.content)
+        .bind(dto.scope.to_string())
```

Comment:
If `ToString` is not required for the `Scope` enum, Im wondering if this could be simplified to:

```suggestion
        .bind(dto.scope)
```